### PR TITLE
Implementation and tests for TPI stats endpoints

### PR DESCRIPTION
**Pull request by conbrad**
- Adds TPI stats endpoint that returns the (assumed) single record for TPI stats for a specific run time and run parameters
    - Since TPI stats are counts of cells that hit HFI > 4k for each classification, this is returned as an area, count * pixel size in metres
- Adds tests for authenticated and unauthenticated endpoint calls
- Removes unnecessary mock JWT with roles in test suite
# Test Links:
[Landing Page](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/morecast)
[Percentile Calculator](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/percentile-calculator)
[C-Haines](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3838-e1e498-dev.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: api/app/routers/fba.py [hunk 1]
```diff
@@ -165,21 +155,28 @@ async def get_run_datetimes_for_date_and_runtype(run_type: RunType, for_date: da
...
+
+@router.get("/fire-zone-tpi-stats/{run_type}/{for_date}/{run_datetime}/{fire_zone_id}", response_model=FireZoneTPIStats)
+async def get_fire_zone_tpi_stats(fire_zone_id: int, run_type: RunType, run_datetime: datetime, for_date: date, _=Depends(authentication_required)):
+    """Return the elevation TPI statistics for each advisory threshold"""
+    logger.info("/fba/fire-zone-tpi-stats/")
+    async with get_async_read_session_scope() as session:
+        stats = await get_zonal_tpi_stats(session, fire_zone_id, run_type, run_datetime, for_date)
+        return FireZoneTPIStats(
+            fire_zone_id=fire_zone_id,
+            valley_bottom=stats.valley_bottom * stats.pixel_size_metres,
```
Comment (dgboss) [inline api/app/routers/fba.py:179]: Is pixel size stored as area? Or a 1-D length? If the latter, then we might want to square the pixel size.

Comment (dgboss) [reply api/app/routers/fba.py:179]: Ultimately it might not matter as we're just displaying percentages in the frontend, not absolute values.

**Review by dgboss — APPROVED** [review]

File: api/app/routers/fba.py (hunk 1 above)
Comment (conbrad) [reply api/app/routers/fba.py]: Done in: https://github.com/bcgov/wps/pull/3838/commits/a8517bc7a28b3dd86e331c69cb8e70f143b1c56f

**Review by brettedw — APPROVED** [review]
Looks great!

**Pull request merged**